project/src/main/kotlin/org/ionproject/core/programme/ProgrammeRepoImpl.kt

```diff
@@ -16,42 +18,66 @@ class ProgrammeRepoImpl(
     private val offerRowReducer: ProgrammeOfferRowReducer
 ) : ProgrammeRepo {
 
-    override fun getProgrammes(): List<Programme> = tm.run { handle ->
-        handle.createQuery(ProgrammeData.GET_PROGRAMMES_QUERY)
+    override fun getProgrammes(page: Int, limit: Int): List<Programme> = tm.run { handle ->
+        val programmes = handle.createQuery(ProgrammeData.GET_PROGRAMMES_QUERY)
+            .bind(ProgrammeData.OFFSET, page * limit)
+            .bind(ProgrammeData.LIMIT, limit)
             .map(programmeMapper)
             .list()
-    } as List<Programme>
 
-    override fun getProgrammeById(id: Int): Programme? = tm.run { handle ->
-        val res = handle.createQuery(ProgrammeData.GET_PROGRAMME_BY_ID_QUERY)
+        if (programmes.isEmpty()) {
+            if (page > 0)
+                throw ResourceNotFoundException("No results for page $page with limit $limit.")
```

Comment:
An empty page is not a `404`. A nonexistent page (e.g. negative index) is a `404`.

Reply:
This is a bigger issue because it's present throughout the codebase. IMO this issue should be created and later changed, to maintain the consistency of the codebase.

Reply:
Also pages with negative indices are IMO a 400 (Bad Request), since an index should always be greater or equal to zero